src/build/buildTask.ts

```diff
@@ -129,12 +127,6 @@ export class BuildTask {
           "-DPYTHON_DEPS_CHECKED=1",
           "-DESP_PLATFORM=1",
         ];
-      } else if (useEqualSign) {
```

Comment:
These changes were intentional and are part of the commit [Remove unnecessary validation - c8f9ec8](https://github.com/espressif/vscode-esp-idf-extension/pull/1271/commits/c8f9ec8efe1ba98b92c84ffee8b00fa0de43ec65)

After some testing, I found out the implementation with "=" was unnecessary. After the changes we've made with the configuration settings. These commands are being run in the python env, so the behaviour is the same for all shells the "=" was added for some specific case Andrii found in the past with consecutive white spaces in the name.

Reply:
I can't see the commit since it is lost maybe due to squash commits.

`These commands are being run in the python env` ? How ? where?

Reply:
I understand your logic but I think is better in the future to separate this into 2 PRs: 1) compile_commands.json work and 2) The removal of `=`. When I see the changes I just thought of it as a regression because it changes code unrelated to the PR.

We can proceed testing it in this PR but then is better to add how is fixed in the PR description and steps to test. 
I think the issue was an old CMake version (Please test with esp-idf v4.3.6) didn't support -S <path> only -S=<path> and it was reported by a customer, hence my remark to make sure this does work.

Reply:
Hi @brianignacio5,

I agree with your vision of splitting PRs, but in this particular case there is one thing that I forgot to mention: which is that I've removed the "=" because there was an issue on this PR that Fabricio found and it was caused by the compiler argument "-B=". 

`Executing task: /home/fabricio/.espressif/python_env/idf5.3_py3.12_env/bin/python /home/fabricio/esp/v5.3.1/esp-idf/tools/idf.py -B=/home/fabricio/Workspace/WSL-Local/blink/build -DSDKCONFIG=/home/fabricio/Workspace/WSL-Local/blink/sdkconfig reconfigure`

![image](https://github.com/user-attachments/assets/35d16784-a5f4-43a5-8540-e8367a214bc9)

I will be updating the testing steps in the description.